infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -125,7 +125,16 @@ public Pod deploy(Pod pod) throws InfrastructureException {
     ObjectMeta metadata = pod.getMetadata();
     PodSpec podSpec = pod.getSpec();
     podSpec.setRestartPolicy("Always"); // Only allowable value
+    Watch toCloseOnException = null;
     try {
+      final CompletableFuture<Pod> createFuture = new CompletableFuture<>();
+      final Watch createWatch =
+          clientFactory
+              .create(workspaceId)
+              .pods()
+              .inNamespace(namespace)
+              .watch(new CreateWatcher(createFuture, workspaceId));
+      toCloseOnException = createWatch;
```

Comment:
Can't we move `createWatch` declaration before try block and not use an additional variable to close it in the finally block?

Reply:
Good catch, I've made the change.